src/server/api/controllers/questions.controller.js

```diff
@@ -3,8 +3,37 @@ const HttpError = require('../lib/utils/http-error');
 // moment is no longer recommended. Should we replace with Luxon? https://moment.github.io/luxon/
 // const moment = require('moment-timezone');
 
+// This function helps with optimisation and cuts out the requirement for a filter function later.
+function groupBy(list, getKey) {
+  const result = {};
+  list.forEach((item) => {
```

Comment:
You can use reduce on lists https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

Reply:
In this case, I thought that forEach was more readable.